bitstreams.md

```diff
@@ -37,14 +37,87 @@ Exposed links:
 
 Bitstream metadata can be modified as described in [Modifying metadata via Patch](metadata-patch.md).
 
+Additional bitstream properties can be modified via Patch as described below.
+
+### Replace
+The replace operation allows to replace *existent* information with new one. Attempt to use the replace operation to set not yet initialized information must return an error. See [general errors on PATCH requests](patch.md)
+
+To change the bundle of a bitstream:
+
+`curl --data '[ { "op": "replace", "path": "/bundleName", "value": "TEST"}]' -H "Authorization: Bearer ..." -H "content-type: application/json" -X PATCH ${dspace7-url}/api/core/bitstreams/${uuid}`
+
+For example, starting with the following bitstream data:
+```json
+{
+  "name" : "test.zip",
+  "bundleName" : "ORIGINAL",
+  "sequenceId" : 5
+}
+```
+the change bundleName operation will result in:
+```json
+{
+  "name" : "test.zip",
+  "bundleName" : "TEST",
+  "sequenceId" : 5
+}
+```
+
+To change the sequenceId:
+
+`curl --data '[ { "op": "replace", "path": "/sequenceId", "value": 2}]' -H "Authorization: Bearer ..." -H "content-type: application/json" -X PATCH ${dspace7-url}/api/core/bitstreams/${uuid}`
+
+
+For example, starting with the following bitstream data:
+```json
+{
+  "name" : "test.zip",
+  "bundleName" : "ORIGINAL",
+  "sequenceId" : 5
+}
+```
+the change sequenceId operation will result in:
+```json
+{
+  "name" : "test.zip",
+  "bundleName" : "ORIGINAL",
+  "sequenceId" : 2
+}
+```
+
+Status codes:
+* 200 OK - if the operation succeed
+* 401 Forbidden - if you are not authenticated
+* 403 Unauthorized - if you are not logged in with sufficient permissions
+* 404 Not found - if the bitstream doesn't exist
+* 422 Unprocessable Entity - if the bundleName was removed, or the sequenceId already exists for the item containing the bitstream
+
 ## Linked entities
 ### Format
-**/api/core/bitstreams/<:uuid>/format**
+**GET /api/core/bitstreams/<:uuid>/format**
 
 Example: <http://dspace7.4science.it/dspace-spring-rest/#http://dspace7.4science.it/dspace-spring-rest/api/core/bitstreams/8d33bdfb-e7ba-43e6-a93a-f445b7e8a1e2/format>
 
 It returns the format of the bitstream
 
+**PUT /api/core/bitstreams/<:uuid>/format**
+
+Update the bitstream format of the bitstream
```

Comment:
I suggest to introduce also a second example using text/plain as format to manage the case of unknown format user defined

Reply:
The unknown format is also an actual format: https://dspace7.4science.cloud/server/#https://dspace7.4science.cloud/server/api/core/bitstreamformats/1
So this shouldn't be any different

Reply:
you are right but there are also a special metadata here involved with the unknown format, see 
https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/content/Bitstream.java#L243

we should decide how to deal with it. Can the user set it updating the metadata? can this be done also when the bitstreamFormat is known? we should include some details about that here in the contract... a solution could be to list this case as a 422 error for updating metadata and leave the /format endpoint only to set the format from the registry
I'm fine with both solution, if we go for the metadata set the unknown format with an user description will require 2 calls instead than 1 but look to me more linear

Reply:
... if we go to use the operation over the metadata to update the format we should also say something about what happen if you change the format from the "unknown format" to a known format. Is it a valid operation that will lead to automatically blank/cleanup the metadata or you need to make two calls otherwise the /format will result in a 422 ?

Reply:
I did notice that the BitstreamService.setFormat https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/content/BitstreamServiceImpl.java#L229 removes the format, even when it's set to unknown.
And setting the dc.format is only possible in the DSpace 6 UI when unknown is the associated type, but it's not mandatory

So to me it makes most sense that:
* setting the format to a known type still removes dc.format
* setting the format to unknown still removes dc.format
* setting dc.format should never implicitly change the format to unknown
* setting dc.format can perhaps be denied when the format is known. But I don't know whether this is relevant enough to create such an exception